### PR TITLE
Remove GetResource and ResourceList message

### PR DESCRIPTION
**Pull request by badb**
Information about resources should be included in ComputeTaskDef. Thanks to that we can:
1) Simplify the protocol
2) Add support for sending different resources for different subtasks, which will be necessary for future use cases.

### CODE REVIEW

File: golem_messages/message/tasks.py [hunk 1]
```diff
@@ -293,6 +294,7 @@ class TaskToCompute(ConcentEnabled, TaskMessage):
         'size',  # the size of the resources zip file
```
Comment (Krigpl) [inline golem_messages/message/tasks.py]: You probably wanna remove this and the line above?

File: tests/message/test_tasks.py [hunk 2]
```diff
@@ -109,6 +109,10 @@ def test_extra_data(self):
         self.assertTrue(extra_data['frames'])
         self.assertTrue(extra_data['output_format'])
 
+    def test_has_resources(self):
+        ctd = message.ComputeTaskDef()
```
Comment (jiivan) [inline tests/message/test_tasks.py]:
```suggestion
        ctd = message.tasks.ComputeTaskDef()
```

Comment (jiivan) [reply tests/message/test_tasks.py]: IMHO This method only tests whether `FrozenDict` works as expected. I think it's not necessary.

File: tests/message/test_tasks.py [hunk 3]
```diff
@@ -348,6 +352,10 @@ def test_size_notint(self):
         with self.assertRaises(exceptions.FieldError):
             helpers.dump_and_load(ttc)
 
+    def test_resources_options(self):
+        ttc = factories.tasks.TaskToComputeFactory()
+        self.assertTrue(hasattr(ttc, 'resources_options'))
```
Comment (jiivan) [inline tests/message/test_tasks.py]: IMHO this method only tests whether `messages.base.Message.load_slots()` works. It initializes all missing slots to `None`. I think it's not necessary to test this in this context.

File: golem_messages/message/tasks.py (hunk 1 above)
Comment (Krigpl) [reply golem_messages/message/tasks.py]: Why not ?

**Pull request merged**